packages/headless/src/features/insight-user-actions/insight-user-actions-preprocessing.test.ts

```diff
@@ -0,0 +1,1202 @@
+            mappedAndSortedActions,
+            ticketCreationDate
+          );
+
+          expect(timeline).toEqual(expectedTimeline);
+        });
+      });
+
+      describe('when the ticket creation date falls between two sessions', () => {
+        it('should return the timeline including current session timestamp set as the ticket creation date', async () => {
+          const mockRawActions = [...fakeActions].slice(0, 8);
+          const mappedAndSortedActions =
+            mapAndSortActionsByMostRecent(mockRawActions);
+          const ticketCreationDate = createRelativeDate(
+            caseCreationDate,
+            120,
+            0
+          ).getTime();
+
+          const sessions = splitActionsIntoTimelineSessions(
+            mappedAndSortedActions,
+            ticketCreationDate
+          );
+
+          expect(sessions.session?.actions.length).toEqual(1);
+          expect(sessions.session?.actions[0].actionType).toEqual(
+            UserActionType.TICKET_CREATION
+          );
+          expect(sessions.session?.actions[0].timestamp).toEqual(
+            ticketCreationDate
+          );
+        });
+      });
+    });
+  });
+
+  describe('#isActionWithinSessionThreshold', () => {
+    it('should return true if the action is within the session inactivity threshold of the latest session action', async () => {
+      const action = {
+        actionType: 'CLICK' as UserActionType,
+        timestamp: firstSessionDate.getTime(),
+        eventData: {},
+        searchHub: 'in-product-help',
+        document: {title: 'title', uriHash: 'TtnKwc0Lo2GY9WAi'},
+      };
+
+      // Added 1 min to action timestamp, so it is within 30 mins of the previous action
+      const previousEndDateTime = createRelativeDate(
+        firstSessionDate,
+        1,
+        0
+      ).getTime();
+
+      const isSameSession = isActionWithinSessionThreshold(
+        action,
+        previousEndDateTime
+      );
+      expect(isSameSession).toEqual(true);
+    });
+
+    it('should return false if the action is not within 30mins of the previous', async () => {
+      const action = {
+        actionType: 'CLICK' as UserActionType,
+        timestamp: firstSessionDate.getTime(),
+        eventData: {},
+        searchHub: 'in-product-help',
+        document: {title: 'title', uriHash: 'TtnKwc0Lo2GY9WAi'},
+      };
+
+      // Added 1 hour to action timestamp, so it is not within 30 mins of the previous action
+      const previousEndDateTime = createRelativeDate(
+        firstSessionDate,
+        60,
+        0
+      ).getTime();
+
+      const isSameSession = isActionWithinSessionThreshold(
+        action,
+        previousEndDateTime
+      );
+      expect(isSameSession).toEqual(false);
+    });
+  });
+
+  describe('#filterTimelineActions', () => {
+    it('should properly filter out the timeline of the actions that are to be excluded and of empty searches', async () => {
```

Comment:
```suggestion
    it('should properly filter out the timeline of custom events that are to be excluded and remove empty searches', async () => {
```

Reply:
removed that test as I moved the filtering from after the timeline was built to before we insert a session in the timeline